### PR TITLE
[student-libraries] Added ability to import one or more libraries into a project

### PR DESCRIPTION
**Pull request by jmkulwik**
# Description
This allows students to import one or more libraries into their project. When a library is imported, the page will refresh so the blocks can be loaded (this is how importing maker toolkit blocks works now). Removing libraries from a project is also enabled as part of this change.

Hidden behind `student-libraries` experiment

Students can import libraries into applab and gamelab.

NOTE: None of this UI is finalized. This is just here as scaffolding for the non-ui processes.
LibrariesImport

## Links

- [spec](https://docs.google.com/document/d/1VxMmdHdIeBWCXKofkjEy9MuWcrzeEW3Zt-Ir8pINYdA/edit)
- [jira](https://codedotorg.atlassian.net/browse/STAR-710)
- [all relevant PRs](https://github.com/pulls?utf8=%E2%9C%93&q=is%3Apr+author%3Ajmkulwik+archived%3Afalse+student+libraries)

## Testing story
Added tests for project.js and StudioApp.js

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/code-studio/components/Libraries/LibraryManagerDialog.jsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
+/*globals dashboard*/
+import PropTypes from 'prop-types';
+import React from 'react';
+import Dialog, {Body} from '@cdo/apps/templates/Dialog';
+import LibraryClientApi from '@cdo/apps/code-studio/components/Libraries/LibraryClientApi';
+
+export default class LibraryManagerDialog extends React.Component {
```
Comment (jmkulwik) [inline apps/src/code-studio/components/Libraries/LibraryManagerDialog.jsx:7]: No tests added here yet because none of this is in its final state

File: apps/src/p5lab/P5Lab.js [hunk 2]
```diff
@@ -272,6 +272,7 @@ P5Lab.prototype.init = function(config) {
 
   config.dropletConfig = dropletConfig;
   config.appMsg = this.isSpritelab ? spritelabMsg : gamelabMsg;
+  this.studioApp_.loadLibraryBlocks(config);
```
Comment (ajpal) [inline apps/src/p5lab/P5Lab.js:275]: Why is this needed? Aren't libraries just in App Lab?

File: apps/src/lib/ui/SettingsCog.jsx [hunk 3]
```diff
@@ -129,6 +138,9 @@ class SettingsCog extends Component {
           showTail={true}
         >
           <ManageAssets onClick={this.manageAssets} />
+          {experiments.isEnabled('student-libraries') && (
```
Comment (ajpal) [inline apps/src/lib/ui/SettingsCog.jsx]: Can you pull the 'student-libraries' experiment string to a constant here: https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/util/experiments.js#L20

File: apps/src/code-studio/components/Libraries/LibraryManagerDialog.jsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      <Dialog isOpen={this.props.isOpen} onCancel={this.props.onClose}>
+        <Body>
+          {this.displayLibraries()}
+          <input
+            type="text"
+            value={this.state.importLibraryId}
+            onChange={this.setLibraryToImport}
+          />
+          <button onClick={this.addLibrary} type="button">
+            Add Library
```
Comment (ajpal) [inline apps/src/code-studio/components/Libraries/LibraryManagerDialog.jsx:76]: i18n?

File: apps/test/unit/StudioAppTest.js [hunk 5]
```diff
@@ -185,6 +186,75 @@ describe('StudioApp', () => {
...
+      level: {
+        codeFunctions: {preExistingFunction: null}
+      },
+      dropletConfig: {
+        additionalPredefValues: ['preExistingValue'],
+        blocks: ['preExistingBlock']
+      }
+    };
+
+    it('given no libraries, leaves the config unchanged', () => {
```
Comment (ajpal) [inline apps/test/unit/StudioAppTest.js:200]: Is this test the same as the next one?

File: apps/src/p5lab/P5Lab.js (hunk 2 above)
Comment (jmkulwik) [reply apps/src/p5lab/P5Lab.js:275]: App Lab and Game Lab

File: apps/test/unit/StudioAppTest.js (hunk 5 above)
Comment (jmkulwik) [reply apps/test/unit/StudioAppTest.js:200]: Oops. Looks like I forgot to update the second one. Thanks for catching that.

File: apps/src/code-studio/components/Libraries/LibraryManagerDialog.jsx (hunk 4 above)
Comment (jmkulwik) [reply apps/src/code-studio/components/Libraries/LibraryManagerDialog.jsx:76]: I'm going to wait on that one because there is a decent chance the final version will have different text or won't be text at all (right now this is just a placeholder). I'll make sure I make a note on this file when I do the UI portion that this file should be evaluated as a while rather than just for changes.

File: apps/src/lib/ui/SettingsCog.jsx (hunk 3 above)
Comment (jmkulwik) [reply apps/src/lib/ui/SettingsCog.jsx]: ✔️

**Review by ajpal — APPROVED** [review]
